internal/controllers/topology/cluster/desired_state.go

```diff
@@ -106,6 +107,15 @@ func (r *Reconciler) computeDesiredState(ctx context.Context, s *scope.Scope) (*
 		}
 	}
 
+	// If required, compute the desired state of the MachinePools from the list of MachinePoolTopologies
```

Comment:
Sorry please also re-diff in this file. We made some significant changes to the MD part

Reply:
Same comment about reusing code